app/services/utils/collection.py

```diff
@@ -0,0 +1,56 @@
+from typing import Optional, Dict, Any
+
+import requests
+
+
+def get_collection_items_url(
+    base_url: str, collection_id: str
+) -> Optional[str]:
+    url = base_url + collection_id
+    response = requests.get(url)
+    response.raise_for_status()
+    collection_data = response.json()
+
+    items_url = None
+    for link in collection_data.get("links", []):
+        if link.get("rel") == "items":
+            items_url = link["href"]
+            break
```

Comment:
again, don't search through the links, but instead verify directly the items url